tests/tests/test_pathfinding.py

```diff
@@ -1,13 +1,135 @@
+        },
+        "schematic": {
+            "type": "LineString",
+            "coordinates": [
+                [-0.387122554630656, 49.4998],
+                [-0.37, 49.4998],
+                [-0.365, 49.4999],
+                [-0.31, 49.4999],
+                [-0.296, 49.4999],
+                [-0.172, 49.4999],
+                [-0.1354, 49.4999],
+                [-0.135, 49.49995],
+                [-0.1346, 49.4999],
+                [-0.12, 49.4999],
+                [-0.115, 49.497],
+                [-0.115, 49.487],
+                [-0.11, 49.484],
+                [-0.095104854807785, 49.484],
+            ],
+        },
+        "steps": [
+            {
+                "track": "TA2",
+                "position": 837.033949007362,
+                "geo": {"type": "Point", "coordinates": [-0.387122554630656, 49.4998]},
+                "sch": {"type": "Point", "coordinates": [-0.387122554630656, 49.4998]},
+                "id": None,
+                "name": None,
+                "suggestion": False,
+                "duration": 0.0,
+            },
+            {
+                "track": "TC2",
+                "position": 450.0,
+                "geo": {"type": "Point", "coordinates": [-0.3037, 49.4999]},
+                "sch": {"type": "Point", "coordinates": [-0.3037, 49.4999]},
+                "id": "Mid_West_station",
+                "name": "Mid_West_station",
+                "suggestion": True,
+                "duration": 0.0,
+            },
+            {
+                "track": "TD1",
+                "position": 14000.0,
+                "geo": {"type": "Point", "coordinates": [-0.22656, 49.4999]},
+                "sch": {"type": "Point", "coordinates": [-0.22656, 49.4999]},
+                "id": "Mid_East_station",
+                "name": "Mid_East_station",
+                "suggestion": True,
+                "duration": 0.0,
+            },
+            {
+                "track": "TH1",
+                "position": 4386.599249046556,
+                "geo": {"type": "Point", "coordinates": [-0.095104854807785, 49.484]},
+                "sch": {"type": "Point", "coordinates": [-0.095104854807785, 49.484]},
+                "id": None,
+                "name": None,
+                "suggestion": False,
+                "duration": 1.0,
+            },
+        ],
+    }
+)
 
-def test_pathfinding_small_infra(small_scenario: Scenario):
-    run_pathfinding(
-        API_URL, small_scenario.infra, [{"track_section": "TE1", "offset": 0}, {"track_section": "TE0", "offset": 0}]
-    )
 
-
-def test_pathfinding_dummy_infra(dummy_scenario: Scenario):
-    run_pathfinding(API_URL, dummy_scenario.infra, _get_waypoints_dummy_infra(API_URL, dummy_scenario.infra))
+def test_west_to_south_east_path(west_to_south_east_path: Path):
+    assert west_to_south_east_path.owner == _EXPECTED_WEST_TO_SOUTH_EAST_PATH.owner
+    assert west_to_south_east_path.slopes == _EXPECTED_WEST_TO_SOUTH_EAST_PATH.slopes
```

Comment:
I don't really like this kind of test with such precise values when we're dealing with floats. Any change (directly or indirectly) related to the way we compute slopes can make the test fail, even if the new result would be just as valid. 

Ideally we'd find a way to add a tolerance to this test, but that can be a pain.

Reply:
I will add [approx](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-approx), they can compare dict and ordered lists it is very powerful. I think it is what tests are supposed to do. Check results consistency and if behavior is changing then test must be updated and it raises a warning to the author and reviewer of the PR. If we want to avoid future regression we have to write tests that are not just checking is status code is 200.